src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -297,7 +313,15 @@ def terminateNodes(self, nodes):
         self._terminateNodes(nodes, self.ctx)
 
     def addNodes(self, nodeType, numNodes, preemptable):
-        instanceType = ec2_instance_types[nodeType]
+        if self.ec2_instance_types:
```

Comment:
This code is duplicated three times. It should go in a function, e.g. getInstanceType.

Reply:
Twice, but good point!  Will fix.